### DISCUSSION TITLE
ChannelMap usage in xslt

### DISCUSSION THREAD

**Discussion by ReneHeuker** [nextgenhealthcare/connect#5652] [Q&A]
I would like to use channelMap variables inside an xslt step, however could nog get it to work. Doen anyone know how this can be done?

**Comment by jonbartels** [comment]
Can you show how you tried to get it to work? Show any errors you got. 

My first guess would be that the XSLT step is able to use E4X escaping so you could have `<xsltThing<{$('myVar')}<xsltThing>`.

the next option is to write your XSLT step, then click on "Generated Script". This is the JS that MC builds from the GUI. You can then copy that JS into a real JS step and modify it using JS to inject your map values.

**Reply by ReneHeuker** [reply]
Thanks you for your answer

I tried all possible ways, but it seems no velocity or anything is done for a xslt template before generating the code. I'll have a look in the mirth code, perhaps I can find anything there. (edit: the code seems not to do anything with the source xslt from the step)

Changing the generated code and add a transformer.setParameter('xxx', $('xxx')); will work but than i can not normally edit my templates anymore

Also another way is to add the channelMap variables to the Source XML and get them from there.  It would have been nice if mirth detected the parameters somehow and add the required transformer.setParameter('xxx', $('xxx')) to the generated code.
